node/config/update.go

```diff
@@ -0,0 +1,239 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+	"unicode"
+
+	"github.com/BurntSushi/toml"
+)
+
+// ConfigUpdate takes in a config and a default config and optionally comments out default values
```

Comment:
It looks like there is a lot of shared code between ConfigUpdate and ConfigDiff.
Can we refactor out the common code into functions called from ConfigUpdate and ConfigDiff?